gateway-service/src/main/java/org/zowe/apiml/gateway/controllers/AuthController.java

```diff
@@ -249,6 +241,25 @@ public Map<String, Object> getCurrentPublicKeys() {
     @ResponseBody
     @HystrixCommand
     public ResponseEntity<Object> getPublicKeyUsedForSigning() {
+        List<JWK> publicKeys = getCurrentKey();
+        if (publicKeys.isEmpty()) {
+            return new ResponseEntity<>(messageService.createMessage("org.zowe.apiml.gateway.keys.unknownState").mapToApiMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+        if (publicKeys.size() != 1) {
+            return new ResponseEntity<>(messageService.createMessage("org.zowe.apiml.gateway.keys.wrongAmount", publicKeys.size()).mapToApiMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
```

Comment:
I think these two conditions should have logs as well, if we return HTTP 500 to the client I think we should have a way to explain the reason from the server side logs

Reply:
logs added